pkg/pipeline/WorkflowDagExecutor.go

```diff
@@ -682,30 +690,60 @@ func (impl *WorkflowDagExecutorImpl) buildWFRequest(runner *pipelineConfig.CdWor
-			return nil, err
-		}
-		pipelineReleaseCounter, err = impl.pipelineOverrideRepository.GetCurrentPipelineReleaseCounter(pipelineId)
-		if err != nil {
-			impl.logger.Errorw("error occurred while fetching latest release counter for pipeline", "pipelineId", pipelineId, "err", err)
-			return nil, err
+	var preDeploySteps []*bean3.StepObject
+	var postDeploySteps []*bean3.StepObject
+	var refPluginsData []*bean3.RefPluginObject
+	//if pipeline_stage present for pre-CD or post-CD then no need to add stageYaml to cdWorkflowRequest in that
+	//case add PreDeploySteps and PostDeploySteps to cdWorkflowRequest, this is done for backward compatibility
+	pipelineStage, err := impl.pipelineStageRepository.GetAllCdStagesByCdPipelineId(cdPipeline.Id)
+	if err != nil {
+		impl.logger.Errorw("error in getting pipelineStages by cdPipelineId", "err", err, "cdPipelineId", cdPipeline.Id)
+		return nil, err
+	}
+	if len(pipelineStage) > 0 {
+		if runner.WorkflowType == bean.CD_WORKFLOW_TYPE_PRE {
+			preDeploySteps, _, refPluginsData, err = impl.pipelineStageService.BuildPrePostAndRefPluginStepsDataForWfRequest(cdPipeline.Id, cdStage)
+			if err != nil {
+				impl.logger.Errorw("error in getting pre, post & refPlugin steps data for wf request", "err", err, "cdPipelineId", cdPipeline.Id)
+				return nil, err
+			}
+		} else if runner.WorkflowType == bean.CD_WORKFLOW_TYPE_POST {
+			_, postDeploySteps, refPluginsData, err = impl.pipelineStageService.BuildPrePostAndRefPluginStepsDataForWfRequest(cdPipeline.Id, cdStage)
+			if err != nil {
+				impl.logger.Errorw("error in getting pre, post & refPlugin steps data for wf request", "err", err, "cdPipelineId", cdPipeline.Id)
+				return nil, err
+			}
+		} else {
+			return nil, fmt.Errorf("unsupported workflow triggerd")
 		}
 
 	} else {
-		return nil, fmt.Errorf("unsupported workflow triggerd")
+		//in this case no plugin script is not present for this cdPipeline hence going with attaching preStage or postStage config
+		if runner.WorkflowType == bean.CD_WORKFLOW_TYPE_PRE {
+			stageYaml = cdPipeline.PreStageConfig
+		} else if runner.WorkflowType == bean.CD_WORKFLOW_TYPE_POST {
+			stageYaml = cdPipeline.PostStageConfig
+			//getting deployment pipeline latest wfr by pipelineId
+			pipelineId := cdPipeline.Id
+			deployStageWfr, err = impl.cdWorkflowRepository.FindLastStatusByPipelineIdAndRunnerType(pipelineId, bean.CD_WORKFLOW_TYPE_DEPLOY)
+			if err != nil {
+				impl.logger.Errorw("error in getting latest status of deploy type wfr by pipelineId", "err", err, "pipelineId", pipelineId)
+				return nil, err
+			}
+			deployStageTriggeredByUser, err = impl.user.GetById(deployStageWfr.TriggeredBy)
+			if err != nil {
+				impl.logger.Errorw("error in getting userDetails by id", "err", err, "userId", deployStageWfr.TriggeredBy)
+				return nil, err
+			}
+			pipelineReleaseCounter, err = impl.pipelineOverrideRepository.GetCurrentPipelineReleaseCounter(pipelineId)
```

Comment:
why don't we need params ``pipelineReleaseCounter``, ``deployStageTriggeredByUser``, ``deployStageWfr`` in case where ``pipelineStage`` is present ??

Reply:
for now have used the same variables in case of POST_CD in pipelineStage section by analogous to previous implementation for post-stage yaml, but need to verify why do we need those at first place, and will take call accordingly.